subsys/net/openthread/rpc/server/ot_rpc_netdiag.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+#include <ot_rpc_ids.h>
+#include <ot_rpc_types.h>
+#include <ot_rpc_common.h>
+
+#include <nrf_rpc_cbor.h>
+
+#include <zephyr/net/openthread.h>
+
+#include <openthread/netdiag.h>
+
+static void ot_rpc_cmd_set_vendor_name(const struct nrf_rpc_group *group,
+				       struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	otError error;
+	char buffer[256];
```

Comment:
Could you use `OT_NETWORK_DIAGNOSTIC_MAX_VENDOR_XXX` constants, instead?

Reply:
as discussed f2f- leaving full buffer on set functions from client->server allows returning invalid arguments error by OT API instead of reporting RPC decoding errors. buffers on getting from server->client can be smaller as we are sure that data saved on server is of correct size